modules/client/components/NumberTextInput.js

```diff
@@ -31,8 +31,10 @@ class NumberTextInput extends React.Component {
     const { value } = this.state
     const displayValue = this.props.formatNumber(value)
```

Comment:
doing `this.props.formatNumber(value || 0)` seems to not be enough, since besides `''`, value could be a non-numeric string.
Since there are no dev-setup instructions, I didn't have an easy way to make further tests.
Anyway, it can be solved either with the proposed fixed, or by ensuring that what `formatNumber` gets passed is indeed a number.